### PR TITLE
[AIRFLOW-5793] add test to detect multiple alembic revision heads

### PR DESCRIPTION
**Pull request by dstandish**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/projects/AIRFLOW/issues/AIRFLOW-5793)
  - https://issues.apache.org/jira/projects/AIRFLOW/issues/AIRFLOW-5793

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
Depending on the timing of merges with migrations, we can end up with two revision heads that need to be merged.

This adds a test to detect when multiple heads are present.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Comment by dstandish** [comment]
@Fokko was unsure about exactly right naming for test file and location.  e.g. could have called it `test_migrations` instead of `test_alembic`.  do lemme know if you think something else is better and i'll update.

File: tests/test_alembic.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+import unittest
+
+from alembic.config import Config
+from alembic.script import ScriptDirectory
+
+
+class TestAlembic(unittest.TestCase):
+
+    def test_only_single_head_revision_in_migrations(self):
```
Comment (Fokko) [inline tests/test_alembic.py]: Maybe move this test to https://github.com/apache/airflow/blob/master/tests/utils/test_db.py

This file contains a closely related test, to check if the schema of the database, and the schema in the python files are in sync.

**Review by Fokko — COMMENTED** [review]
Love it, one comment

File: tests/test_alembic.py (hunk 1 above)
Comment (Fokko) [reply tests/test_alembic.py]: Maybe also add a small comment on when we could have two heads, and how to resolve it :-)

Comment (feluelle) [reply tests/test_alembic.py]: I think this would be better to have in the PR template? (and also in the test)

Comment (potiuk) [reply tests/test_alembic.py]: I think the PR template is already pretty bloated :) (I think we should somehow simplify this). I found it initially useful, but then it becomes more and more tiresome. Let's not make it more complex.

I think everything that is now in the template should be automatically verified by the checks in CI and commiters should not even think about checking it.

Comment (potiuk) [reply tests/test_alembic.py]: Agree with Fokko -> better have less test files.

Comment (feluelle) [reply tests/test_alembic.py]: > I think the PR template is already pretty bloated :) (I think we should somehow simplify this). I found it initially useful, but then it becomes more and more tiresome. Let's not make it more complex.

Totally agree. Would be nice if we the template changes depending on the files that changed.

Comment (Fokko) [reply tests/test_alembic.py]: Can you rebase onto master and move this test to the other file? :-)

Comment (dstandish) [reply tests/test_alembic.py]: Yes will do very shortly was out for a bit

Comment (dstandish) [reply tests/test_alembic.py]: moved test.  added small comment, just pointing user to `alembic merge` command.

**Review by Fokko — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Comment by Fokko** [comment]
Thanks @dstandish

**Comment by kaxil** [comment]
Thanks @dstandish

**Pull request merged**